desktop/apps/personalize/components/App.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import { ContextProvider } from '@artsy/reaction-force/dist/Components/Artsy'
+import Wizard from '@artsy/reaction-force/dist/Components/Onboarding/Wizard'
+import CollectorIntent from '@artsy/reaction-force/dist/Components/Onboarding/Steps/CollectorIntent'
+import Artists from '@artsy/reaction-force/dist/Components/Onboarding/Steps/Artists'
+import Genes from '@artsy/reaction-force/dist/Components/Onboarding/Steps/Genes'
+import Budget from '@artsy/reaction-force/dist/Components/Onboarding/Steps/Budget'
+const sd = require('sharify').data
```

Comment:
Good thinking @alloy ! I have no clue (although this looks like client-side code?).

On the server, as correctly done in the `routes.js` file below (https://github.com/artsy/force/pull/2096/files#diff-93fdfd7268bf104e320c6480d6844a64R8), we should always do `res.locals.sd.CURRENT_USER` server-side.

Reply:
@alloy @mzikherman this gets executed on both server + client. Trying to understand the flaw though...when I print out `sd` from this file on server side, it only gives me the keys we know to be "public", ie [these](https://github.com/artsy/force/blob/master/lib/setup_sharify.js). At that point, sd.CURRENT_USER is actually `null` which I think is ok for server-side since we don't need any user info until the client loads. 

Why do we need to use res.locals.sd.CURRENT_USER for server side again?

Reply:
Also happy to make all this more clear by passing in CURRENT_USER as a prop from `res.locals.sd` but I still want to understand.

Reply:
I second @kanaabe's comment. I'm not sure what the vulnerability is?

Reply:
The issue is `CURRENT_USER` is more of a request-level thing, whereas using `require('sharify').data` on the server means, in the case of concurrent requests, that you might not be getting what you expect.

I see here that this is just going to be returning `null`, so this is _probably_ fine, but theoretically, you should be using sharify on the server via `res.locals.sd....`, so it feels weird to see `require('sharify').data`, _especially_ for `CURRENT_USER`, which is naturally a request-level attribute (unlike some more global config).

Reply:
Yes, sorry, by ‘safe’ I meant that it is not safe to think that this will give you the current user, which, as @mzikherman points out, is a request-level concern.

I get what you’re saying, though, that you aren’t actually expecting it to contain anything on the server-side, but a naive reader (such as yours truly) could easily look at this file and be led astray. In the future that could even lead to actual incorrect decisions being made, which is just my paranoia speaking, especially when it comes to concurrent code.

In short, I think code should be self-documenting as much as possible and in this case explicitly passing that data in is from that perspective probably more clear.

Reply:
Gotcha, thanks for your explanations! Updating this in a bit 👍